x/ecocredit/server/basket/take_test.go

```diff
@@ -172,6 +173,42 @@ func TestTakeTradable(t *testing.T) {
 	s.expectRetiredSupply("C4", "0")
 }
 
+func TestTakeAllTradable(t *testing.T) {
+	t.Parallel()
+	s := setupTake(t)
+
+	barCoins := sdk.NewCoins(sdk.NewCoin("bar", sdk.NewInt(11000000))) // == 7C4 + 4C4
```

Comment:
Do we need an invariant to check, for each basket denom, sum(BankKeeper balances) == sum(Ecocredit balances)